keychain_linux.cpp

```diff
@@ -111,16 +117,15 @@ std::string getPassword(const std::string &package, const std::string &service,
                                                 user.c_str(),
                                                 NULL);
 
+    updateError(err, error);
```

Comment:
This one missing is a major bug. I couldn't come up with a regression test though, because I don't know how I can make `secret_password_lookup_sync` fail :(